HpToolsLauncher/Runners/AlmTestSetsRunner.cs

```diff
@@ -993,32 +994,29 @@ private void SetApiTestParameters(ITSTest3 test, string paramsString)
         /// <param name="paramsString"></param>
         private void SetGuiTestParameters(ITSTest3 test, string paramsString)
         {
-            var xmlParameters = new StringBuilder();
-            List<string> parameterNames = new List<string>();
-            List<string> parameterValues = new List<string>();
+            var xmlParams = new StringBuilder();
+            IList<string> paramNames = new List<string>(), paramValues = new List<string>();
 
-            if (!string.IsNullOrEmpty(paramsString))
+            if (!string.IsNullOrWhiteSpace(paramsString))
             {
-                string[] parameters = paramsString.Split(',');
+                string[] @params = paramsString.Split(COMMA, StringSplitOptions.RemoveEmptyEntries);
 
-                bool validParameters = ValidateListOfParameters(paramsString, parameters, parameterNames, parameterValues);
+                bool validParameters = ValidateListOfParams(paramsString, @params, ref paramNames, ref paramValues);
 
-                if (validParameters)
+                if (validParameters && @params.Any())
                 {
-                    xmlParameters.Append("<?xml version=\"1.0\"?><Parameters>");
-                    for (int i = 0; i < parameters.Length; i++)
+                    xmlParams.Append(XML_PARAMS_START_TAG);
+                    for (int i = 0; i < @params.Length; i++)
                     {
-                        xmlParameters.Append("<Parameter><Name><![CDATA[{parameterNames.ElementAt(i)}]]></Name>")
-                                     .AppendFormat("<Value><![CDATA[{0}]]>", parameterValues.ElementAt(i))
-                                     .Append("</Value></Parameter>");
+                        xmlParams.AppendFormat(XML_PARAM_NAME_VALUE, paramNames[i], paramValues[i]);
```

Comment:
The `paramValues[i]` may contain the sequence `]]>` when the value is a valid XML data. It might be better to escape the sequence: simply replace `]]>` with `]]]]><![CDATA[>`. Reference: https://stackoverflow.com/questions/223652/is-there-a-way-to-escape-a-cdata-end-token-in-xml

Reply:
Good point, I notified the QA team to test + rise a defect for this scenario.